.github/src/readme.js

```diff
@@ -0,0 +1,33 @@
+// @ts-check
+
+import { marked } from "marked";
+import yaml from "yaml";
+
+/**
+ * @param {string} markdown
+ * @param {import('github-script').AsyncFunctionArguments['core']} core
+ * @returns {Promise<Set<string>>} All input files for all tags
+ */
+export async function getInputFiles(markdown, core) {
+  const tokens = marked.lexer(markdown);
+
+  const yamlBlocks = tokens
+    .filter((token) => token.type === "code")
+    .map((token) => /** @type import("marked").Tokens.Code */ (token))
+    // Include default block and tagged blocks (```yaml $(tag) == 'package-2021-11-01')
```

Comment:
Eventually we will also need to do some basic condition handling because when reading the yaml content in a readme.md you need to know what chunks are applicable. I have a hacky version of this I implemented in PS at https://github.com/Azure/azure-sdk/blob/main/eng/scripts/Update-Spec-Versions.ps1#L266 for doing spec discovery. 

I eventually would like to use the code we build up in this repo do to that spec discovery so we have one consistent way of doing it.

Reply:
I agree, but I'd like to wait to add these helpers until we need them.  Fitering yaml blocks by tag should be fairly easy.